tests/tests.rs

```diff
@@ -85,24 +85,31 @@ fn run_bindgen_tests() {
         }
     });
 
-    // First spawn all child processes and collect them, then wait on each
-    // one. This runs the tests in parallel rather than serially.
-
-    let children: Vec<_> = tests.map(|entry| {
+    let children = tests.map(|entry| {
             let child = spawn_run_bindgen(run_bindgen.clone(), bindgen.clone(), entry.path());
             (entry.path(), child)
-        })
-        .collect();
-
-    let failures: Vec<_> = children.into_iter()
-        .filter_map(|(path, mut child)| {
-            let passed = child.wait()
-                .expect("Should wait on child process")
-                .success();
-
-            if passed { None } else { Some((path, child)) }
-        })
-        .collect();
+        });
+
+    let child_wait = |(path, mut child): (PathBuf, process::Child)| {
```

Comment:
Can you make this a proper nested function (`fn child_wait((path, mut child): (PathBuf, process::Child)) -> Option<(PathBuf, process::Child)> { ... }`) instead, to reflect it's not capturing any state from the environment?